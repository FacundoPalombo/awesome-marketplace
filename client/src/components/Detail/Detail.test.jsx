import { render } from "@testing-library/jest-dom";

const mockItem = {
  id: "MLA880793636",
  title: "Xiaomi Redmi 9 (global) Dual Sim 64 Gb Carbon Grey 4 Gb Ram",
  price: 34950,
  condition: "new",
  sold_quantity: 50,
  categories: [
    {
      id: "MLA1051",
      name: "Celulares y Teléfonos",
    },
    {
      id: "MLA1055",
      name: "Celulares y Smartphones",
    },
  ],
  description:
    'Enfocate en lo importante\nEl novedoso sistema operativo Android 10 incorpora respuestas inteligentes y acciones sugeridas para todas tus aplicaciones. Además, incluye la función de Bienestar Digital y el Tema Oscuro, para que evites distracciones y logres una mayor concentración.\n\nMayor rendimiento\nSu memoria RAM de 4 GB te permitirá ejecutar varias aplicaciones al mismo tiempo, jugar y navegar con gran rapidez y sin inconvenientes.\n\nMás para ver\nCon su pantalla IPS de 6.53", disfrutá de colores intensos y mayor nitidez en todos tus contenidos.\n\nGran capacidad de almacenamiento\nCon su memoria interna de 64 GB siempre tendrás espacio para almacenar archivos y documentos importantes. Además, podrás guardar películas, series y videos para reproducirlos cuando quieras sin conexión.\n\nBatería superior\n¡Desenchufate! Con la súper batería de 5020 mAh, tendrás energía por mucho más tiempo para jugar, ver series o trabajar sin necesidad de recargar tu teléfono.\n\nFotografía profesional en tu bolsillo\nDescubrí infinitas posibilidades para tus fotos con las 4 cámaras principales de tu equipo. Gracias a la cámara teleobjetivo capturarás detalles casi imperceptibles, con la de ángulo amplio sacarás fotos nítidas y la ultra gran angular te permitirá obtener imágenes panorámicas excepcionales. ¿Amás los fondos difuminados? Vas a conseguirlos con el famoso modo retrato de la cuarta cámara.\n\n Además, el dispositivo cuenta con cámara frontal de 8 Mpx para que puedas sacarte divertidas selfies o hacer videollamadas.\n\nTecnología premium\nMaximizá tu seguridad y asegurate de que solo vos puedas desbloquear el equipo. Gracias al sensor de huella dactilar, podrás habilitar tu dispositivo con solo un toque. Además, cuenta con reconocimiento facial que se activa rápidamente al colocar la pantalla frente a tu rostro.',
};
