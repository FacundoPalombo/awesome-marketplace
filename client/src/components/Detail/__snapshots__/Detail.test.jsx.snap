// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Detail Component WHEN detail was retrieved AND product is new THEN should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <img
        class="thumbnail"
      />
      <h4
        class="descriptionTitle"
      >
        Descripción del producto
      </h4>
      <p
        class="description"
      />
    </div>
    <div
      class="heading"
    >
      <p
        class="condition"
      >
        Usado -  vendidos
      </p>
      <h4
        class="title"
      />
      <h3
        class="price"
      />
      <button
        class="cta"
      >
        Comprar
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN Detail Component WHEN detail was retrieved AND product is used THEN should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <img
        class="thumbnail"
      />
      <h4
        class="descriptionTitle"
      >
        Descripción del producto
      </h4>
      <p
        class="description"
      />
    </div>
    <div
      class="heading"
    >
      <p
        class="condition"
      >
        Nuevo -  vendidos
      </p>
      <h4
        class="title"
      />
      <h3
        class="price"
      />
      <button
        class="cta"
      >
        Comprar
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN Detail Component WHEN results was NOT retrieved THEN should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <img
        class="thumbnail"
      />
      <h4
        class="descriptionTitle"
      >
        Descripción del producto
      </h4>
      <p
        class="description"
      />
    </div>
    <div
      class="heading"
    >
      <p
        class="condition"
      >
        Usado -  vendidos
      </p>
      <h4
        class="title"
      />
      <h3
        class="price"
      />
      <button
        class="cta"
      >
        Comprar
      </button>
    </div>
  </div>
</DocumentFragment>
`;
